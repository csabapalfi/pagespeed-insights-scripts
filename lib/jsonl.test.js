const {results, stats} = require('./jsonl');

describe('jsonl formatter', () => {
  it('results are JSON.stringify-ed', () => {
    expect(results([{
      type: 'result',
      fetchTime: '21:48:05',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    }])).toMatchSnapshot();
  });

  it('stats are JSON.stringify-ed', () => {
    expect(stats([{
      type: 'statistic',
      name: 'stddev',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    }])).toMatchSnapshot();
  });
});