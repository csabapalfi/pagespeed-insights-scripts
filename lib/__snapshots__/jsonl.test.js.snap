// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsonl formatter results are JSON.stringify-ed 1`] = `
Array [
  "{\\"type\\":\\"result\\",\\"fetchTime\\":\\"21:48:05\\",\\"score\\":95,\\"FCP\\":0.9,\\"FMP\\":1,\\"SI\\":1.1,\\"FCI\\":3.2,\\"TTI\\":4}",
]
`;

exports[`jsonl formatter stats are JSON.stringify-ed 1`] = `
Array [
  "{\\"type\\":\\"statistic\\",\\"name\\":\\"stddev\\",\\"score\\":95,\\"FCP\\":0.9,\\"FMP\\":1,\\"SI\\":1.1,\\"FCI\\":3.2,\\"TTI\\":4}",
]
`;
